test/excel.test.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* eslint-env mocha */
+const assert = require('assert');
+const { main } = require('../src/index');
+
+describe('Excel Tests', () => {
+  it('Works for Excel Feeds', async () => {
+    const result = await main({
+      __ow_path: '/https://adobe.sharepoint.com/sites/TheBlog/_layouts/15/guestaccess.aspx',
+      share: 'ESR1N29Z7HpCh1Zfs_0YS_gB4gVSuKyWRut-kNcHVSvkew',
+      email: 'helix%40adobe.com',
```

Comment:
what is email used for?

Reply:
It's not needed, but part of the share link as you copy it from the UI.

Reply:
as I said, better provide the entire sharelink as param.

Reply:
I don't get it. The entire share link is `https://adobe.sharepoint.com/sites/TheBlog/_layouts/15/guestaccess.aspx?share=ESR1N29Z7HpCh1Zfs_0YS_gB4gVSuKyWRut-kNcHVSvkew&email=helix%40adobe.com&e=rcfCT3`

This gets appended to the service URL, so that the path is `/https://adobe.sharepoint.com/sites/TheBlog/_layouts/15/guestaccess.aspx` and the URL parameters are passed as individual parameters. 

If I pass the remaining parameters as `__ow_query="share=ESR1N29Z7HpCh1Zfs_0YS_gB4gVSuKyWRut-kNcHVSvkew&email=helix%40adobe.com&e=rcfCT3"` I'm not gaining anything, because I now have to parse the query string – something that only occurs in testing, never in production.

Reply:
my problem is, that your sharelink argument is distributed as path and query params. this is inconsistent and pollutes the action params. I would do:

`...data-embed@v1?share=https://adobe.sharepoint.com/sites/TheBlog/_layouts/15/guestaccess.aspx?share=ESR1N29Z7HpCh1Zfs_0YS_gB4gVSuKyWRut-kNcHVSvkew&email=helix%40adobe.com&e=rcfCT`

Reply:
I'd rather keep it the way it is, so that `helix-embed` and `helix-data-embed` have the same API (just append the full URL)

Reply:
I don' think this is good. if the url would be url-encoded, i would be marginally better.
you already see from your code, that you need to reconstruct the share link based on the path and some params...and you always need to rely on the exact format of the link.

what if MS or google decides to add a new query param to the sharelink?